docs/OpenSSLSupport.md

```diff
@@ -148,6 +148,31 @@ Note that the code snippet for the RDRAND engine opt-in is required to use not o
+This ensures that when an enclave includes a patched OpenSSL header and uses the specific API or macro, user will receive compile-time errors.
+The errors can be disabled by specifying the `OE_OPENSSL_SUPPRESS_UNSUPPORTED` option to the compiler.
+
+See the following table for the detailed list of APIs/Macros.
+
+API / Macro | Original header | Comments | Guidance |
+:---:|:---:|:---|:---|
+OPENSSL_INIT_LOAD_CONFIG | crypto.h | The macro represents an option to the OPENSSL_init_ssl and OPENSSL_init_crypto APIs that initializes an application based on the openssl.cnf (loaded from the host filesystem). Therefore, the use of this option would allow an untrusted host to fully control the initialization of the application. | The recommendation is not to invoke initialization APIs with this option. Note that starting from v1.1.0, the explicit initialization is not required. |
+SSL_CTX_load_verify_locations | ssl.h | The API specifies the default locations from which an application looks up CA certificates for verification purposes. The API would allow an untrusted host to fully control what certificates the application will trust. | The recommendation is using the SSL_CTX_set_cert_store API that specifies an in-memory certificate verification storage (`X509_STORE`). Another alternative is to implement a customized verification callback and sets the callback via the SSL_CTX_set_verify API, which effectively bypasses the default implementation. |
+SSL_CTX_set_default_verify_paths | ssl.h | The API specifies the locations as part of arguments from which an application looks up CA certificates for verification purposes. The API would allow an untrusted host to fully control what certificates the application will trust. | Same recommendation as SSL_CTX_load_verify_locations. |
+SSL_CTX_set_default_verify_dir | ssl.h | Similar to SSL_CTX_set_default_verify_paths except for specifying dir only. | Same recommendation as SSL_CTX_load_verify_locations. |
+SSL_CTX_set_default_verify_file | ssl.h | Similar to SSL_CTX_set_default_verify_paths except for specifying file only. | Same recommendation as SSL_CTX_load_verify_locations. |
+X509_LOOKUP_hash_dir | x509_vfy.h | This API returns a `X509_LOOKUP_METHOD` method that loads files from the path specified by the `SSL_CERT_DIR` environment variable. The API is used by SSL_CTX_set_default_verify_dir and SSL_CTX_set_default_verify_paths internally. | The recommendation is implementing a customized `X509_LOOKUP_METHOD` method based on in-memory operations. |
+X509_LOOKUP_file | x509_vfy.h | This API returns a `X509_LOOKUP_METHOD` method that loads files from the path specified by the `SSL_CERT_FILE` environment variable. The API is used by SSL_CTX_set_default_verify_file and SSL_CTX_set_default_verify_paths internally. | The recommendation is not to use this API (i.e., the lookup method associated with it). | Same recommendation as X509_LOOKUP_hash_dir. |
```

Comment:
This row has an extra column, so the Guidance ends up displaying as

> The recommendation is not to use this API (i.e., the lookup method associated with it).

Instead of the intended:

> Same recommendation as X509_LOOKUP_hash_dir.

Also, for this and `X509_STORE_set_default_paths`, I would suggest just copy-pasting the guidance rather than having the reader go hunt in the table for the reference.

Reply:
Thanks for the catch!